### PR TITLE
add mold v2.2.0; bump dependencies

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **mold/all**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mold/all/conanfile.py [hunk 1]
```diff
@@ -45,12 +45,12 @@ def package_id(self):
         del self.info.settings.compiler
 
     def requirements(self):
-        self.requires("zlib/1.2.13")
-        self.requires("openssl/1.1.1q")
-        self.requires("xxhash/0.8.1")
-        self.requires("onetbb/2021.3.0")
+        self.requires("zlib/1.3")
+        self.requires("openssl/1.1.1w")
```
Comment (SpaceIm) [inline recipes/mold/all/conanfile.py:49]: Version range for these dependencies maybe?

Comment (AndreyMlashkin) [reply recipes/mold/all/conanfile.py]: done

**Comment by AndreyMlashkin** [comment]
Is it a CI-related fail?

**Comment by AndreyMlashkin** [comment]
blocked by https://github.com/conan-io/conan-center-index/pull/20813

**Comment by fschoenm** [comment]
@AndreyMlashkin Your `license` is wrong (see #19152). I don't know if you want to integrate my changes into your branch (I'm waiting for more than two months now for somebody to fix the _onetbb_ problem) or wait for my PR to be merged before bumping the version.

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin Your `license` is wrong (see #19152). I don't know if you want to integrate my changes into your branch (I'm waiting for more than two months now for somebody to fix the _onetbb_ problem) or wait for my PR to be merged before bumping the version.

Thanks for heads up, I cherry-picked your commit to my branch now :)

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 @SpaceIm 
could someone please generate binaries for onetbb?

ERROR: Missing binary: onetbb/2021.3.0:d7a9f73b4f7a501a35ebea3234d4801bf5740350

**Comment by uilianries** [comment]
@AndreyMlashkin I just triggered a new build for OneTBB 👍

**Review by SSE4 — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
> @AndreyMlashkin I just triggered a new build for OneTBB 👍

still
ERROR: Missing binary: onetbb/2021.3.0:d7a9f73b4f7a501a35ebea3234d4801bf5740350

**Comment by uilianries** [comment]
@AndreyMlashkin The PR https://github.com/conan-io/conan-center-index/pull/20897 is under discussion to fix this situation. I'll talk to Conan team about it, because there are possible side-effects. Sorry the inconvenience

**Comment by valgur** [comment]
The `onetbb` issues should be resolved now. Try building again @AndreyMlashkin.

**Comment by valgur** [comment]
Also mold v2.4.0 is out. You should bump the version in this PR, if possible.

**Comment by jcar87** [comment]
Closed as superseded by https://github.com/conan-io/conan-center-index/commit/53813eb7e80298e3e407bb5563d01cff612a5d88

but please let us know if version `2.2.0` specifically needs to be covered and we'll consider it.

**Pull request closed without merging**